src/main/scala/rugds/sbt/DockerDevEnvironment.scala

```diff
@@ -0,0 +1,177 @@
+          case (module, dependencies) => module -> dependencies.filter(_.jarFile.isDefined)
+        }
+      jarDependencies.map {
+        case (module, dependencies) => if (module == projectId) module -> ((jarDependencies.toMap.keySet ++ dependencies) - module) else module -> dependencies
+      }.sortWith {
+        case ((m1, d1), (m2, d2)) => d2.contains(m1) || !d1.contains(m2)
+      }
+    },
+
+    envDockerDependencyTree := envProjectDependencyTree.value.flatMap {
+      case (module, dependencies) => module.toEnvDocker.map(_ -> dependencies.flatMap(_.toEnvDocker))
+    },
+
+    envPullDockerImages := {
+      val log = streams.value.log
+      envDockerDependencyTree.value.map {
+        case (image, _) => image.toString
+      }
+      .map { image => {
+        log.info(s"docker images -q $image")
+        if (s"docker images -q $image".!!.isEmpty) {
+          log.info(s"docker pull $image")
+          if (s"docker pull $image".! != 0) {
+            throw new IllegalStateException(s"Cannot found or pull docker image: $image")
+          }
+        }
+        image
+      }}.toSet
+    },
+
+    envStartDev := {
+      val log = streams.value.log
+      val pdi = envPullDockerImages.value
+//      val networkCmd = s"docker -P" // TODO: auto-create network if not yet created (or fail if exists?)
+      val networkId = "rugds-dev"
+      envDockerDependencyTree.value.map {
+        case (docker, dependencies) =>
+          val dockerPsCmd = Seq("docker", "ps", "-aq", "--format", "{{ .Names}}"); log.info(dockerPsCmd.mkString(" "))
+          if (!dockerPsCmd.lines.contains(docker.name)) { // if it does not exist => run it
+            val dockerInspectCmd = Seq("docker", "inspect", "-f", "{{range $key, $item := .Config.ExposedPorts}}{{$key}} {{end}}", docker.toString); log.info(dockerInspectCmd.mkString(" "))
+            val dockerExposedPorts = dockerInspectCmd.!!
+            val exposedPorts = dockerExposedPorts.trim.split(' ').map(port => port.split('/')).map(list => (list.head, list.last)).map {
+              case (port, _) => s"-p $port:$port"
+            }.mkString(" ")
+            val dockerRunCmd = s"docker run -d --network=$networkId $exposedPorts --name ${docker.name} ${docker.toString}"; log.info(dockerRunCmd)
+            dockerRunCmd.!!
+          }
+          docker
+      }.toSet
+    },
+
+    envDockerList := (resources in Compile).value
+      .filter(_.isDirectory)
+      .filter(_.name == "docker-dependencies")
+      .flatMap(_.listFiles)
+      .flatMap(repo => {
+        repo.listFiles.map(image => {
+          val imageVersion = scala.io.Source.fromFile(image.getAbsolutePath + "/version").mkString
+          val dockerBuild  = if (new File(image.getAbsolutePath + "/Dockerfile").exists()) Some(image) else None
+          EnvDocker(repo.name, image.name, imageVersion, dockerBuild)
+        })
+      })
+      .filter(_.file.isDefined)
+      .toSet,
+
+    envDockerBuild := envDockerList.value
```

Comment:
See above comment, does this only build images for current project, or also for external